lib/chef/knife/vsphere_datastore_list.rb

```diff
@@ -46,27 +29,47 @@ class Chef::Knife::VsphereDatastoreList < Chef::Knife::BaseVsphereCommand
          short: '-L',
          description: "Indicates whether to list VM's in datastore",
          boolean: true
+  option :pool,
+         long: '--pool pool',
+         short: '-h',
```

Comment:
I grabbed this from the [hosts_list](https://github.com/chef-partners/knife-vsphere/blob/master/lib/chef/knife/vsphere_hosts_list.rb#L13) command options to be consistent. I just did some testing and it looks like it works just fine. 

if i move the `-h` before `vsphere`  then the help is displayed

`knife -h vsphere datastore list --list -h MYCLUSTER`

I cna change this to `-p` if it makes more sense though?